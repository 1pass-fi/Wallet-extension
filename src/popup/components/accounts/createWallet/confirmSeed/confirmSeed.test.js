import '@babel/polyfill'
import { fireEvent, render, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import React from 'react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { BrowserRouter as Router } from 'react-router-dom'
import thunk from 'redux-thunk'

jest.mock('lodash/shuffle', () => (list) => list)

import ConnectToWallet from '.'

const mockStore = configureStore([thunk])

describe('Test for ConnectToWallet component', () => {
  let setErrorMock, storeMock, container, setCreateWalletMock, saveWalletMock

  beforeEach(() => {
    setErrorMock = jest.fn()
    setCreateWalletMock = jest.fn()
    saveWalletMock = jest.fn()

    const initState = {
      setError: setErrorMock,
      setCreateWallet: setCreateWalletMock,
    }

    storeMock = mockStore(initState)

    container = render(
      <Router>
        <Provider store={storeMock}>
          <ConnectToWallet
            password='12345678'
            seedPhrase='fragile narrow pepper warfare aisle average gospel danger quarter warm egg civil'
            saveWallet={saveWalletMock}
          />
        </Provider>
      </Router>
    ).container
  })

  describe('renders without crashing', () => {
    it('renders correctly', () => {
      expect(container).toMatchSnapshot()
    })
  })

  describe('select word in phrase', () => {
    it('renders correctly', () => {
      const unselectedWord = document.querySelector('.unselected-box .word')
      fireEvent.click(unselectedWord)
      expect(container).toMatchSnapshot()
      const selectedWord = document.querySelector('.selected-box .word')
      fireEvent.click(selectedWord)
      expect(container).toMatchSnapshot()
      const submitButton = document.querySelector('.confirm-button')
      fireEvent.click(submitButton)
      expect(container).toMatchSnapshot()
      const cancelButton = document.querySelector('.cancel-icon')
      fireEvent.click(cancelButton)
      expect(container).toMatchSnapshot()
    })
  })

  describe('select all words in phrase', () => {
    it('renders correctly', () => {
      const unselectedWords = document.querySelectorAll('.unselected-box .word')
      for (let i = 0; i < 12; i += 1) {
        fireEvent.click(unselectedWords[i])
      }
      const submitButton = document.querySelector('.confirm-button')
      fireEvent.click(submitButton)
      expect(saveWalletMock).toHaveBeenCalled()
      expect(container).toMatchSnapshot()
    })
  })

  describe('select all words in phrase', () => {
    beforeEach(() => {
      saveWalletMock.mockImplementation(() => {
        throw new Error()
      })
    })

    it('renders correctly', () => {
      const unselectedWords = document.querySelectorAll('.unselected-box .word')
      for (let i = 0; i < 12; i += 1) {
        fireEvent.click(unselectedWords[i])
      }
      const submitButton = document.querySelector('.confirm-button')
      fireEvent.click(submitButton)
      expect(saveWalletMock).toHaveBeenCalled()
    })
  })
})
