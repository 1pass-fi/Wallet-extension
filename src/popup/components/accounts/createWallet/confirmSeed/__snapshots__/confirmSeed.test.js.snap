// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for ConnectToWallet component renders without crashing renders correctly 1`] = `
<div>
  <div
    class="confirmation-card-wrapper"
  >
    <div
      class="card confirmation-card"
    >
      <div
        class="cancel-icon"
      >
        <svg />
      </div>
      <div
        class="title"
      >
        <svg
          class="icon"
        />
        <div
          class="text"
        >
          Secret Backup Phrase
        </div>
      </div>
      <div
        class="description"
      >
        Your secret backup phrase makes it easy to back up and restore your account.
      </div>
      <div
        class="warning"
      >
        <svg
          class="warning-icon"
        />
        <div
          class="warning-text"
        >
          Your secret backup phrase makes it easy to back up and restore your account.
        </div>
      </div>
      <div
        class="selected-box"
      />
      <div
        class="unselected-box"
      >
        <button
          class="word "
        >
          fragile
        </button>
        <button
          class="word "
        >
          narrow
        </button>
        <button
          class="word "
        >
          pepper
        </button>
        <button
          class="word "
        >
          warfare
        </button>
        <button
          class="word "
        >
          aisle
        </button>
        <button
          class="word "
        >
          average
        </button>
        <button
          class="word "
        >
          gospel
        </button>
        <button
          class="word "
        >
          danger
        </button>
        <button
          class="word "
        >
          quarter
        </button>
        <button
          class="word "
        >
          warm
        </button>
        <button
          class="word "
        >
          egg
        </button>
        <button
          class="word "
        >
          civil
        </button>
      </div>
      <button
        class="button-shared button-deactivated confirm-button"
      >
        Confirm
      </button>
    </div>
  </div>
</div>
`;

exports[`Test for ConnectToWallet component select all words in phrase renders correctly 1`] = `
<div>
  <div
    class="confirmation-card-wrapper"
  >
    <div
      class="card confirmation-card"
    >
      <div
        class="cancel-icon"
      >
        <svg />
      </div>
      <div
        class="title"
      >
        <svg
          class="icon"
        />
        <div
          class="text"
        >
          Secret Backup Phrase
        </div>
      </div>
      <div
        class="description"
      >
        Your secret backup phrase makes it easy to back up and restore your account.
      </div>
      <div
        class="warning"
      >
        <svg
          class="warning-icon"
        />
        <div
          class="warning-text"
        >
          Your secret backup phrase makes it easy to back up and restore your account.
        </div>
      </div>
      <div
        class="selected-box"
      >
        <button
          class="word"
        >
          fragile
        </button>
        <button
          class="word"
        >
          narrow
        </button>
        <button
          class="word"
        >
          pepper
        </button>
        <button
          class="word"
        >
          warfare
        </button>
        <button
          class="word"
        >
          aisle
        </button>
        <button
          class="word"
        >
          average
        </button>
        <button
          class="word"
        >
          gospel
        </button>
        <button
          class="word"
        >
          danger
        </button>
        <button
          class="word"
        >
          quarter
        </button>
        <button
          class="word"
        >
          warm
        </button>
        <button
          class="word"
        >
          egg
        </button>
        <button
          class="word"
        >
          civil
        </button>
      </div>
      <div
        class="unselected-box"
      >
        <button
          class="word disabled"
        >
          fragile
        </button>
        <button
          class="word disabled"
        >
          narrow
        </button>
        <button
          class="word disabled"
        >
          pepper
        </button>
        <button
          class="word disabled"
        >
          warfare
        </button>
        <button
          class="word disabled"
        >
          aisle
        </button>
        <button
          class="word disabled"
        >
          average
        </button>
        <button
          class="word disabled"
        >
          gospel
        </button>
        <button
          class="word disabled"
        >
          danger
        </button>
        <button
          class="word disabled"
        >
          quarter
        </button>
        <button
          class="word disabled"
        >
          warm
        </button>
        <button
          class="word disabled"
        >
          egg
        </button>
        <button
          class="word disabled"
        >
          civil
        </button>
      </div>
      <button
        class="button-shared button-filled confirm-button"
      >
        Confirm
      </button>
    </div>
  </div>
</div>
`;

exports[`Test for ConnectToWallet component select word in phrase renders correctly 1`] = `
<div>
  <div
    class="confirmation-card-wrapper"
  >
    <div
      class="card confirmation-card"
    >
      <div
        class="cancel-icon"
      >
        <svg />
      </div>
      <div
        class="title"
      >
        <svg
          class="icon"
        />
        <div
          class="text"
        >
          Secret Backup Phrase
        </div>
      </div>
      <div
        class="description"
      >
        Your secret backup phrase makes it easy to back up and restore your account.
      </div>
      <div
        class="warning"
      >
        <svg
          class="warning-icon"
        />
        <div
          class="warning-text"
        >
          Your secret backup phrase makes it easy to back up and restore your account.
        </div>
      </div>
      <div
        class="selected-box"
      >
        <button
          class="word"
        >
          fragile
        </button>
      </div>
      <div
        class="unselected-box"
      >
        <button
          class="word disabled"
        >
          fragile
        </button>
        <button
          class="word "
        >
          narrow
        </button>
        <button
          class="word "
        >
          pepper
        </button>
        <button
          class="word "
        >
          warfare
        </button>
        <button
          class="word "
        >
          aisle
        </button>
        <button
          class="word "
        >
          average
        </button>
        <button
          class="word "
        >
          gospel
        </button>
        <button
          class="word "
        >
          danger
        </button>
        <button
          class="word "
        >
          quarter
        </button>
        <button
          class="word "
        >
          warm
        </button>
        <button
          class="word "
        >
          egg
        </button>
        <button
          class="word "
        >
          civil
        </button>
      </div>
      <button
        class="button-shared button-deactivated confirm-button"
      >
        Confirm
      </button>
    </div>
  </div>
</div>
`;

exports[`Test for ConnectToWallet component select word in phrase renders correctly 2`] = `
<div>
  <div
    class="confirmation-card-wrapper"
  >
    <div
      class="card confirmation-card"
    >
      <div
        class="cancel-icon"
      >
        <svg />
      </div>
      <div
        class="title"
      >
        <svg
          class="icon"
        />
        <div
          class="text"
        >
          Secret Backup Phrase
        </div>
      </div>
      <div
        class="description"
      >
        Your secret backup phrase makes it easy to back up and restore your account.
      </div>
      <div
        class="warning"
      >
        <svg
          class="warning-icon"
        />
        <div
          class="warning-text"
        >
          Your secret backup phrase makes it easy to back up and restore your account.
        </div>
      </div>
      <div
        class="selected-box"
      />
      <div
        class="unselected-box"
      >
        <button
          class="word "
        >
          fragile
        </button>
        <button
          class="word "
        >
          narrow
        </button>
        <button
          class="word "
        >
          pepper
        </button>
        <button
          class="word "
        >
          warfare
        </button>
        <button
          class="word "
        >
          aisle
        </button>
        <button
          class="word "
        >
          average
        </button>
        <button
          class="word "
        >
          gospel
        </button>
        <button
          class="word "
        >
          danger
        </button>
        <button
          class="word "
        >
          quarter
        </button>
        <button
          class="word "
        >
          warm
        </button>
        <button
          class="word "
        >
          egg
        </button>
        <button
          class="word "
        >
          civil
        </button>
      </div>
      <button
        class="button-shared button-deactivated confirm-button"
      >
        Confirm
      </button>
    </div>
  </div>
</div>
`;

exports[`Test for ConnectToWallet component select word in phrase renders correctly 3`] = `
<div>
  <div
    class="confirmation-card-wrapper"
  >
    <div
      class="card confirmation-card"
    >
      <div
        class="cancel-icon"
      >
        <svg />
      </div>
      <div
        class="title"
      >
        <svg
          class="icon"
        />
        <div
          class="text"
        >
          Secret Backup Phrase
        </div>
      </div>
      <div
        class="description"
      >
        Your secret backup phrase makes it easy to back up and restore your account.
      </div>
      <div
        class="warning"
      >
        <svg
          class="warning-icon"
        />
        <div
          class="warning-text"
        >
          Your secret backup phrase makes it easy to back up and restore your account.
        </div>
      </div>
      <div
        class="selected-box"
      />
      <div
        class="unselected-box"
      >
        <button
          class="word "
        >
          fragile
        </button>
        <button
          class="word "
        >
          narrow
        </button>
        <button
          class="word "
        >
          pepper
        </button>
        <button
          class="word "
        >
          warfare
        </button>
        <button
          class="word "
        >
          aisle
        </button>
        <button
          class="word "
        >
          average
        </button>
        <button
          class="word "
        >
          gospel
        </button>
        <button
          class="word "
        >
          danger
        </button>
        <button
          class="word "
        >
          quarter
        </button>
        <button
          class="word "
        >
          warm
        </button>
        <button
          class="word "
        >
          egg
        </button>
        <button
          class="word "
        >
          civil
        </button>
      </div>
      <button
        class="button-shared button-deactivated confirm-button"
      >
        Confirm
      </button>
    </div>
  </div>
</div>
`;

exports[`Test for ConnectToWallet component select word in phrase renders correctly 4`] = `
<div>
  <div
    class="confirmation-card-wrapper"
  >
    <div
      class="card confirmation-card"
    >
      <div
        class="cancel-icon"
      >
        <svg />
      </div>
      <div
        class="title"
      >
        <svg
          class="icon"
        />
        <div
          class="text"
        >
          Secret Backup Phrase
        </div>
      </div>
      <div
        class="description"
      >
        Your secret backup phrase makes it easy to back up and restore your account.
      </div>
      <div
        class="warning"
      >
        <svg
          class="warning-icon"
        />
        <div
          class="warning-text"
        >
          Your secret backup phrase makes it easy to back up and restore your account.
        </div>
      </div>
      <div
        class="selected-box"
      />
      <div
        class="unselected-box"
      >
        <button
          class="word "
        >
          fragile
        </button>
        <button
          class="word "
        >
          narrow
        </button>
        <button
          class="word "
        >
          pepper
        </button>
        <button
          class="word "
        >
          warfare
        </button>
        <button
          class="word "
        >
          aisle
        </button>
        <button
          class="word "
        >
          average
        </button>
        <button
          class="word "
        >
          gospel
        </button>
        <button
          class="word "
        >
          danger
        </button>
        <button
          class="word "
        >
          quarter
        </button>
        <button
          class="word "
        >
          warm
        </button>
        <button
          class="word "
        >
          egg
        </button>
        <button
          class="word "
        >
          civil
        </button>
      </div>
      <button
        class="button-shared button-deactivated confirm-button"
      >
        Confirm
      </button>
    </div>
  </div>
</div>
`;
